### PR TITLE
New serverless pattern - s3 - lambda - bedrock

### PR DESCRIPTION
**Pull request by mfwahid**
*Issue #, if available:*

*Description of changes:*
This is a new pattern to integrate S3 to Lambda to Bedrock. This pattern is useful for LLM RAG Usecases in getting vector embeddings of the documents uploaded to S3 via S3 event notification. Once the files are placed in S3, AWS Lambda picks up the files via event notification and post the content to Amazon Bedrock embedding model and gets the vector embedding of the content as response.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by mfwahid** [comment]
Hi, 

Reaching out to see if there's any action item on me? 

Thanks
Abdul Waheed

File: s3-lambda-bedrock/README.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    * Enter the desired AWS Region
+    * Allow SAM CLI to create IAM roles with the required permissions.
+
+    Once you have run `sam deploy --guided` mode once and saved arguments to a configuration file (samconfig.toml), you can use `sam deploy` in future to use these defaults.
+
+1. Note the outputs from the SAM deployment process. These contain the resource names and/or ARNs which are used for testing.
+
+## How it works
+
+Explain how the service interaction works.
```
Comment (mavi888) [inline s3-lambda-bedrock/README.md:42]: can you add the information?

File: s3-lambda-bedrock/README.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    ```
+1. From the command line, use AWS SAM to deploy the AWS resources for the pattern as specified in the template.yml file:
+    ```
+    sam build
+    sam deploy --guided
+    ```
+1. During the prompts:
+    * Enter a stack name
+    * Enter the desired AWS Region
+    * Allow SAM CLI to create IAM roles with the required permissions.
```
Comment (mavi888) [inline s3-lambda-bedrock/README.md:34]: it asks for a content bucket, if this is not automatically created please add it to the requirements. If not create it as IaC in the template

File: s3-lambda-bedrock/README.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+1. Note the outputs from the SAM deployment process. These contain the resource names and/or ARNs which are used for testing.
+
+## How it works
+
+Explain how the service interaction works.
+
+## Testing
+
+Create a text document with some content and upload it to the S3 bucket (bucket details in sam deploy output). You will be able to validate the lambda response in Cloudwatch Logs LogGroup.
```
Comment (mavi888) [inline s3-lambda-bedrock/README.md:46]: can you add an example file, the command to upload a file to s3 using cli
and the commands in cli to see the logs for lambda, so everything can be done from the terminal.

Comment (mavi888) [reply s3-lambda-bedrock/README.md:46]: also dont forget to mention that the file should be in teh directory content
also add what you should see? The function executes, but what should be expecting from this funciton?

**Comment by mavi888** [comment]
please improve the readme, and make sure that you add a purpose to the funciton. now it just prints in console a call to bedrock.

File: s3-lambda-bedrock/README.md (hunk 1 above)
Comment (mfwahid) [reply s3-lambda-bedrock/README.md]: added and commited

File: s3-lambda-bedrock/README.md (hunk 2 above)
Comment (mfwahid) [reply s3-lambda-bedrock/README.md:35]: modified and committed

**Comment by mavi888** [comment]
Thanks for submitting your pattern.
Now this pattern is public in Serverless land

https://serverlessland.com/patterns/s3-lambda-bedrock

You can share your pattern using this link :)

**Pull request merged**